### PR TITLE
Monet theming

### PR DESCRIPTION
**Pull request by elliotnash**
Add support to source bluebubbles' colors from monet colors (Material You dynamic colors) on supported devices running Android 12+

### CODE REVIEW

File: lib/layouts/settings/chat_list_panel.dart [hunk 1]
```diff
@@ -34,7 +34,8 @@ class ChatListPanel extends StatelessWidget {
       headerColor = Theme.of(context).backgroundColor;
       tileColor = Theme.of(context).colorScheme.secondary;
     }
-    if (SettingsManager().settings.skin.value == Skins.iOS && isEqual(Theme.of(context), oledDarkTheme)) {
+    if (SettingsManager().settings.skin.value == Skins.iOS && Theme.of(context).backgroundColor == Colors.black
```
Comment (zlshames) [inline lib/layouts/settings/chat_list_panel.dart:37]: Hey can u make all these booleans into a helper? It's a lot of the same reused code and we should make a helper function for it

**Review by zlshames — CHANGES_REQUESTED** [review]

File: lib/layouts/settings/theme_panel.dart [hunk 2]
```diff
@@ -252,6 +253,27 @@ class ThemePanel extends GetView<ThemePanelController> {
                 SettingsSection(
                   backgroundColor: tileColor,
                   children: [
+                    if (!kIsWeb && !kIsDesktop && monetPalette!=null)
```
Comment (zlshames) [inline lib/layouts/settings/theme_panel.dart:256]: Spacing around != Please

File: lib/helpers/themes.dart [hunk 3]
```diff
@@ -165,11 +168,31 @@ void loadTheme(BuildContext? context, {ThemeObject? lightOverride, ThemeObject?
   if (context == null) return;
 
   // Set the theme to match those of the settings
-  ThemeObject light = lightOverride ?? ThemeObject.getLightTheme();
-  ThemeObject dark = darkOverride ?? ThemeObject.getDarkTheme();
+  ThemeData light = (lightOverride ?? ThemeObject.getLightTheme()).themeData;
+  ThemeData dark = (darkOverride ?? ThemeObject.getDarkTheme()).themeData;
+
+  if (SettingsManager().settings.monetTheming.value && monetPalette!=null) {
```
Comment (zlshames) [inline lib/helpers/themes.dart:174]: Spacing around != Please

**Comment by zlshames** [comment]
Just a few small things. But other than that, the PR looks good. Thanks for using the existing system for themeing. 

Just a follow-up, does this support if your background changes while the app is open? Like does it change if your colors change while the app is open?

**Comment by elliotnash** [comment]
Thanks, I'll make those changes when I get a chance.

The theme is updated when the system color is changed (without any work needing to be done), google just restarts every running app when you change the colors.

**Pull request merged**